core/scripts/server.ts

```diff
@@ -280,47 +282,43 @@ app.get('*', (req, res, next) => {
         } else {
           res.setHeader('X-VS-Cache', 'Miss')
           console.log(`cache miss [${req.url}], request: ${Date.now() - s}ms`)
-          dynamicRequestHandler(renderer) // render response
+          dynamicRequestHandler(renderer, config) // render response
         }
       }).catch(errorHandler)
     } else {
-      dynamicRequestHandler(renderer)
+      dynamicRequestHandler(renderer, config)
     }
   }
 
-  if (config.server.dynamicConfigReload) {
-    const cachedConfigModule = require.cache[require.resolve('config')]
-    if (cachedConfigModule) {
-      delete cachedConfigModule.parent.children
-      delete require.cache[require.resolve('config')]
-    }
-    config = require('config') // reload config
+  let requestContextConfig: any = config.util.extendDeep({}, config);
+
+  if (config.server.dynamicConfigReload && !globalContextConfig) {
     if (configProviders.length > 0) {
+      const configPromises = [];
       configProviders.forEach(configProvider => {
         if (typeof configProvider === 'function') {
-          configProvider(req).then(loadedConfig => {
-            config = config.util.extendDeep(config, loadedConfig)
-            dynamicCacheHandler()
-          }).catch(error => {
-            if (config.server.dynamicConfigContinueOnError) {
-              dynamicCacheHandler()
-            } else {
-              console.log('config provider error:', error)
+          configPromises.push(configProvider(req).then(loadedConfig => {
+            requestContextConfig = config.util.extendDeep(requestContextConfig, loadedConfig)
+          }).catch(() => {
+            if (!config.server.dynamicConfigContinueOnError) {
               if (req.url !== '/error') {
                 res.redirect('/error')
               }
-              dynamicCacheHandler()
             }
-          })
+          }))
         }
       })
-    } else {
-      config = require('config') // reload config
-      dynamicCacheHandler()
+      await Promise.all(configPromises)
+
+      if (!config.server.dynamicConfigReloadWithEachRequest) {
```

Comment:
I understand it as if it is true then config will be reloaded with each request. But we check here negation and then reload config. So shouldn't it be 
```
      if (config.server.dynamicConfigReloadWithEachRequest) {
        globalContextConfig = config.util.extendDeep({}, requestContextConfig)
      }
```
and do we need extra config ? 🤔 Shouldn't `dynamicConfigReload ` handle that?